curvefs/src/metaserver/storage/rocksdb_options.cpp

```diff
@@ -0,0 +1,182 @@
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+/*
+ * Project: curve
+ * Date: Thursday Jun 09 10:34:12 CST 2022
+ * Author: wuhanqing
+ */
+
+#include "curvefs/src/metaserver/storage/rocksdb_options.h"
+
+#include <gflags/gflags.h>
+
+#include <memory>
+#include <mutex>
+
+#include "curvefs/src/metaserver/storage/rocksdb_storage.h"
+#include "rocksdb/filter_policy.h"
+#include "rocksdb/table.h"
+#include "src/common/gflags_helper.h"
+
+namespace curvefs {
+namespace metaserver {
+namespace storage {
+
+// TODO(wuhanqing): block cache and writer buffer manager's capacity should be
+//                  related to the number of copysets
+DEFINE_int64(rocksdb_block_cache_capacity, 8ULL << 30, "xxx");
+
+DEFINE_int64(rocksdb_write_buffer_manager_capacity, 6ULL << 30, "xxx");
+
+// https://github.com/facebook/rocksdb/wiki/Write-Buffer-Manager
+DEFINE_bool(rocksdb_WBM_cost_block_cache,
+            true,
+            "Control wether write buffer manager cost block cache");
+
+DEFINE_int32(
+    rocksdb_max_background_jobs,
+    16,
+    "Maximum number of concurrent background jobs (compactions and flushes)");
+
+DEFINE_double(rocksdb_memtable_prefix_bloom_size_ratio, 0.1, "xxx");
+
+DEFINE_int64(rocksdb_unordered_cf_write_buffer_size, 64 << 20, "xxx");
+
+DEFINE_int32(rocksdb_unordered_cf_max_write_buffer_number, 2, "xxx");
+
+DEFINE_int64(rocksdb_ordered_cf_write_buffer_size, 64 << 20, "xxx");
+
+DEFINE_int32(rocksdb_ordered_cf_max_write_buffer_number, 2, "xxx");
+
+namespace {
+
+std::shared_ptr<rocksdb::Cache> rocksdbBlockCache;
+std::shared_ptr<rocksdb::WriteBufferManager> rocksdbWriteBufferManager;
+
+const RocksDBStorageComparator rocksdbStorageComparator;
+
+const char* const kOrderedColumnFamilyName = "ordered_column_family";
+
+void CreateBlockCacheAndWriterBufferManager() {
+    static std::once_flag createBlockCache;
+    std::call_once(createBlockCache, []() {
+        rocksdbBlockCache =
+            rocksdb::NewLRUCache(FLAGS_rocksdb_block_cache_capacity);
+    });
+
+    static std::once_flag createWriterBufferManager;
+    std::call_once(createWriterBufferManager, []() {
+        rocksdbWriteBufferManager =
+            std::make_shared<rocksdb::WriteBufferManager>(
+                FLAGS_rocksdb_write_buffer_manager_capacity,
+                FLAGS_rocksdb_WBM_cost_block_cache ? rocksdbBlockCache
+                                                   : nullptr);
+    });
+}
+
+}  // namespace
+
+void InitRocksdbOptions(
+    rocksdb::DBOptions* options,
+    std::vector<rocksdb::ColumnFamilyDescriptor>* columnFamilies) {
+    CreateBlockCacheAndWriterBufferManager();
+
+    options->create_if_missing = true;
```

Comment:
Why are the parameters here mostly hardcoded in `InitRocksdbOptions`

Reply:
for most of them, we won't change, so there's no need to put them into configuration file.